Framework/Algorithms/src/GetAllEi.cpp

```diff
@@ -222,8 +223,8 @@ void GetAllEi::exec() {
   auto monitorWS = buildWorkspaceToFit(inputWS, det1WSIndex);
 
   // recalculate delay time from chopper position to monitor position
-  auto detector1 = inputWS->getDetector(det1WSIndex);
-  double mon1Distance = detector1->getDistance(*moderator);
+  auto detectorPosition = inputWS->spectrumInfo().position(det1WSIndex);
+  double mon1Distance = detectorPosition.distance(moderator->getPos());
```

Comment:
Do we have to same problem with the distance calculation for a detector group here as discussed yesterday in `GetEi2`?